metadata-service/factories/src/main/java/com/linkedin/metadata/boot/steps/WaitForBuildIndicesStep.java

```diff
@@ -18,7 +18,7 @@ public String name() {
 
   @Override
   public void execute() throws Exception {
-    if (!_buildIndicesKafkaListener.waitForBootstrap()) {
+    if (!_dataHubUpgradeKafkaListener.waitForBootstrap()) {
```

Comment:
Awesome!

I like the ideal of BootstrapDependency. Thinking about how we can use the same abstraction in the MySQL based UpgradeStep interface... 

It's nice that the step need not know much about what's happening inside of the dependency.